builder/src/upload.tsx

```diff
@@ -232,12 +243,24 @@ const SubmissionForm: React.FC<SubmissionFormProps> = observer((props) => {
         } catch (e) {
             Sentry.captureException(e);
             const errors = new FormErrors();
-            errors.generalError = "Unknown error occurred while uploading file";
+            errors.generalErrors.push(
+                "Unknown error occurred while uploading file"
+            );
             setFormErrors(errors);
             console.error(e);
         }
     };
 
+    const hasErrors =
+        (fileUpload?.uploadErrors ?? []).length > 0 ||
+        formErrors.generalErrors.length > 0;
+
+    const hasEtagError =
+        fileUpload &&
+        fileUpload.uploadErrors.filter(
```

Comment:
`.some()` would be a bit more efficient than looping over the whole array with `.filter()`.

Reply:
Yep, forgot about that existing, was looking for `.any()` and that didn't exist

Reply:
Changed